gateway/src/test/java/io/zeebe/gateway/topology/TopologyUpdateTest.java

```diff
@@ -174,6 +177,52 @@ public void shouldUpdateTopologyOnBrokerRemoveAndDirectlyRejoin() {
     assertThat(topologyManager.getTopology().getLeaderForPartition(1)).isEqualTo(leaderId);
   }
 
+  @Test
+  public void shouldUpdateTopologyOnPartitionHealth() {
+    final int brokerId = 0;
+    final int partition = 0;
+    final BrokerInfo broker = createBroker(brokerId);
+    broker.setPartitionHealthy(partition);
+    topologyManager.event(createMemberAddedEvent(broker));
+    waitUntil(() -> topologyManager.getTopology() != null);
+    assertThat(topologyManager.getTopology().isPartitionHealthy(brokerId, partition)).isTrue();
```

Comment:
Topology does not have partition health in this version. You can remove this test.

Reply:
I'm surprised the cherry pick included this test as well, I don't think I had added it as part of the original PR? :thinking:

Reply:
I have also seen that behavior before. This https://stackoverflow.com/a/19001323 explains it. I did not expect this semantics of cherry-pick.